### PR TITLE
add tests for controller_manager scripts

### PR DESCRIPTION
**Pull request by mathias-luedtke**
addresses #287 

I have added some basic tests without argument tests.
To make the tests work I had to fix a bug in controller_manager list.

In addtion I have refactored the `unspawner` script.
This alters the behavior a little bit, but prevents any infinite blocking.

### CODE REVIEW

**Comment by mathias-luedtke** [comment]
The lunar test fails because the spawner hits a [rospy bug](https://github.com/ros/ros_comm/pull/1026) which does not handle certain error cases correctly.
#284 should fix it at our end.

**Comment by bmagyar** [comment]
Is this still a WIP? My merging finger is itchy! :)

**Review by bmagyar — APPROVED** [review]

**Comment by mathias-luedtke** [comment]
I am not sure about the `unspawner`, I don't know if this is used by someone at all..

**Comment by bmagyar** [comment]
I used it on some PAL robots, it's  probably still that way.

File: controller_manager_tests/test/controller_manager_scripts.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        rospy.wait_for_service("/controller_manager/list_controllers", 2.0)
+
+        run_cm('list-types') # TODO: check output
+        # spawn controller
+        s1 = subprocess.Popen('rosrun controller_manager spawner my_controller __name:=myspawner', shell=True)
+        rospy.sleep(1.0)
+
+        self.assertEqual(run_cm('list'), myc_running)
+
+        run_cm('load', 'my_controller2') # TODO: check output
```
Comment (bmagyar) [inline controller_manager_tests/test/controller_manager_scripts.py]: Aren't these TODO-s supposed to be done too? ;)

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: controller_manager_tests/test/controller_manager_scripts.py (hunk 1 above)
Comment (bmagyar) [reply controller_manager_tests/test/controller_manager_scripts.py]: @ipa-mdl Any notes on this?

**Comment by bmagyar** [comment]
Ping @ipa-mdl

**Comment by mathias-luedtke** [comment]
I might fix the TODOs after the holidays..

**Comment by mikepurvis** [comment]
LGTM

**Comment by mathias-luedtke** [comment]
updated to check all ouput
@bmagyar: please review again

**Review by bmagyar — APPROVED** [review]

**Pull request merged**